packages/x-date-pickers/src/internals/components/PickersPopper.tsx

```diff
@@ -112,13 +112,19 @@ const PickersPopperPaper = styled(MuiPaper, {
   overridesResolver: (_, styles) => styles.paper,
 })<{
   ownerState: PickersPopperOwnerState;
-}>(({ ownerState }) => ({
+}>({
   outline: 0,
   transformOrigin: 'top center',
-  ...(ownerState.placement.includes('top') && {
-    transformOrigin: 'bottom center',
-  }),
-}));
+  variants: [
+    {
+      props: ({ ownerState }: { ownerState: PickersPopperOwnerState }) =>
```

Comment:
Nitpick: If need be, those 3 ('top', 'top-end', 'top-start') cases could be listed individually. 💡 
Or this could be a possible optimization for the variants support, like:
```ts
{
  props : { placement: ['top', 'top-end', 'top-start'] },
  style : { /*...*/ },
}
```

Reply:
Do you mean doing the following?

```diff
- props: ({ placement }: PickersPopperOwnerState) => placement.includes('top'),
+ props: ({ placement }: PickersPopperOwnerState) => ['top', 'top-end', 'top-start'].includes(placement),
```

If you think it's more readable I can do the change :+1:

Reply:
No clear preference. 🤔 
If both ways TS will catch any problems, then all checks out. 👍